src/lib/ngsi/ContextElement.cpp

```diff
@@ -154,7 +157,7 @@ void ContextElement::fill(const struct ContextElement& ce)
   attributeDomainName.fill(ce.attributeDomainName);
   contextAttributeVector.fill((ContextAttributeVector*) &ce.contextAttributeVector);
   domainMetadataVector.fill((MetadataVector*) &ce.domainMetadataVector);
-  providingApplication = ce.providingApplication;  
+  providingApplicationList = ce.providingApplicationList;  // FIXME P10: not sure if this is the right way of doing, taking into account how the same is done for contextAttributeVector in its own fill() method...
```

Comment:
For complex cases (vector which elements are objects that could have vectors in sequences) I agree that implementing ourselves will be better. However, in this case `providingApplicationList` is just a vector of std::string.

Looking to http://www.cplusplus.com/reference/vector/vector/operator=/ it seems that vector contents are copied. Thus, I think it should work. However, I'd add a comment in the code about this.